src/OrleansAzureUtils/Hosting/ServiceRuntimeWrapper.cs

```diff
@@ -183,11 +183,9 @@ private void Initialize()
                 logger.Warn(ErrorCode.AzureServiceRuntime_NotLoaded, msg1);
 
                 // Microsoft.WindowsAzure.ServiceRuntime isn't loaded. We may be running within a web role or not in Azure.
-                // Trying to load by partial name, so that we are not version specific.
-                // Assembly.LoadWithPartialName has been deprecated. Is there a better way to load any version of a known assembly?
-                #pragma warning disable 618
-                assembly = Assembly.LoadWithPartialName("Microsoft.WindowsAzure.ServiceRuntime");
-                #pragma warning restore 618
+#pragma warning disable 618
+                assembly = Assembly.Load(new AssemblyName("Microsoft.WindowsAzure.ServiceRuntime"));
+#pragma warning restore 618
```

Comment:
No not referring to a specific version, but to try load the one version which exists in the app directory through brutal force try load, since LoadWithPartialName is obsolete now. 

The code would look like:
```
List<string> serviceRuntimeFullNameOfAllVersions = []...
foreach(var serviceRuntimeFullname in serviceRuntimeFullNameOfAllVersions )
{
try
{
  Assembly.Load(new AssemblyName(serviceRuntimeFullname ));
  break;
}catch(Exception e)
{
log(e);
}
}
```